### DISCUSSION TITLE
How do I pull up DB info based on the logged in user with dynamic gets ?

### DISCUSSION THREAD

**Discussion by jeejeestudio** [joomengine/Joomla-Component-Builder#962] [Q&A] [answered]
Hi,

I know how to get the ID of the current user in Joomla e.g. echo JText::sprintf('Hello %s', $this->user->id);  
and normally with php I know how to get records from DB based on that ID.
But how do I do that with Dynamic gets ?

Do I need to use Tweak, Joint or Custom script php before getting items?
Its not clear for me. A simple example would be great.

Thank you.

**Comment by Llewellynvdm** [comment, marked as answer]
> I know how to get the ID of the current user in Joomla e.g. echo \JText::sprintf('Hello %s', $this->user->id);

Well this will depend on where your code is being run, since the model, view and controller all have their unique way of giving you the user ID, and it will also depend on which Joomla version you are targeting. I would consider this a very broad question and hard to give one simple answer. Ideally those who use JCB will build their component then open the file/class where they want to do custom code, and look at what is available to them. They will also add some identifiable placeholder so they can see where the custom code from the GUI of JCB gets used in the code it generates. Generally this take time and is hard to explain since there are literally hundreds of use-cases all over JCB with many different ways, and that mostly since this is how Joomla in those areas use to work, or still works.

That all said in the dynamicGet area we are working in the model, and here we have some helper values in (targeting) Joomla 4 and 5 these values gets you the current user ID. [Here you can see the template file used](https://github.com/vdm-io/Joomla-Component-Builder/blob/3.x/admin/compiler/joomla_4/CUSTOM_ADMIN_VIEW_MODEL.php#L147). So you will see a list of values we load by default in the model for use in your code. So to get the user ID in the model `$this->userId`

>  how to get records from DB based on that ID.

I am not sure how many of the [JCB tutorials](https://www.youtube.com/playlist?list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE) => [wiki](https://git.vdm.dev/joomla/Component-Builder/wiki) you've done, but I can tell you that they are still your best source of knowledge about JCB even though they are old.

Assuming you've put in the time... here is a few screenshots of how its done:
![image](https://github.com/vdm-io/Joomla-Component-Builder/assets/5607939/6ac12b45-2ed1-4593-9d05-815bdebd98c8)
![image](https://github.com/vdm-io/Joomla-Component-Builder/assets/5607939/908b3281-0c90-4ea5-8c14-b4c8503e89d0)
![image](https://github.com/vdm-io/Joomla-Component-Builder/assets/5607939/2bcc83dc-39d9-44e7-b904-14e163a49f18)
![image](https://github.com/vdm-io/Joomla-Component-Builder/assets/5607939/c758d662-acc4-4e14-9d36-70145cb0c745)

Hope this points you in the right direction. Take care!

**Reply by jeejeestudio** [reply]
Thanks!
I will test it out.
Of course I also need to think out of the box, meaning I can alter the scripts created by JCB.

**Reply by Llewellynvdm** [reply]
Well yes, you can.... but JCB is not like other wire-framing scripts, it is a all rounder component builder and maintainer. So staying in JCB is what is the real power of JCB. Like we have just moved many components from Joomla 3 to Joomla 4 and 5 in just a few weeks with JCB. This means round-trip-development with JCB is the key objective.

JCB has become the developers tool to build, maintain and release multiple extension targeting different versions of Joomla.

So while you can of course leave the JCB instance and continue with your extension in your normal IDE. This is not how many of us are using JCB... we have continued to maintain our extension inside JCB for years... and it has saved us hundreds of hours.

Interesting video to watch is: [Round Trip Development Demonstration](https://www.youtube.com/watch?v=QdGWXNLnmkk)

**Reply by jeejeestudio** [reply]
Ok, then I think I partly misunderstood what you meant.
Of course I would like to do it all from within JCB, I've tried the old way too. But this is the fastest way.
I check out some other videos of you on Youtube.
Thanks!
